components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsResourceImplTest.java

```diff
@@ -0,0 +1,199 @@
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.query.internal.DefaultQuery;
+import org.xwiki.rest.XWikiRestException;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.test.mockito.MockitoComponentMockingRule;
+import org.xwiki.users.User;
+import org.xwiki.users.UserManager;
+
+import javax.inject.Provider;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.slf4j.Logger;
+
+public class DefaultPatientsResourceImplTest {
+
+    @Rule
+    public MockitoComponentMockingRule<PatientsResource> mocker =
+        new MockitoComponentMockingRule<PatientsResource>(DefaultPatientsResourceImpl.class);
+
+    @Mock
+    private User currentUser;
+
+    @Mock
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private QueryManager queries;
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private EntityReferenceResolver<EntityReference> currentResolver;
+
+    private DomainObjectFactory factory;
+
+    @Mock
+    private Patient patient;
+
+    private URI uri;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    private DefaultPatientsResourceImpl patientsResource;
+
+    private XWikiContext context;
+
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException {
+        MockitoAnnotations.initMocks(this);
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager compManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(compManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.patientsResource = (DefaultPatientsResourceImpl)this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.queries = this.mocker.getInstance(QueryManager.class);
+        this.uri = new URI("http://uri");
+
+        doReturn(this.uri).when(this.uriInfo).getBaseUri();
+        doReturn(this.uri).when(this.uriInfo).getRequestUri();
+        ReflectionUtils.setFieldValue(this.patientsResource, "uriInfo", this.uriInfo);
+
+        doReturn("00000001").when(this.patient).getId();
+        doReturn(this.currentUser).when(this.users).getCurrentUser();
+        doReturn(null).when(this.currentUser).getProfileDocument();
```

Comment:
Return a real DocumentReference object here and mock the `currentResolver` to return a real DocumentReference object and check for them when mocking the behaviour of `hasAccess` instead of checking for null values or mocks. 
i.e: 
```java 
private DocumentReference userProfileDocument = new DocumentReference("wiki", "user", "00000001");
```